File: src/java/SQ_Energy_Balance/EnergybalanceState.java
import  java.io.*;
import  java.util.*;
import java.time.LocalDateTime;
public class EnergyBalanceState
{
    private Double diffusionLimitedEvaporation;
    private Double conductance;
    private Double minCanopyTemperature;
    private Double maxCanopyTemperature;
    
    public EnergyBalanceState() { }
    
    public EnergyBalanceState(EnergyBalanceState toCopy, boolean copyAll) // copy constructor 
    {
        if (copyAll)
        {
            this.diffusionLimitedEvaporation = toCopy.getdiffusionLimitedEvaporation();
            this.conductance = toCopy.getconductance();
            this.minCanopyTemperature = toCopy.getminCanopyTemperature();
            this.maxCanopyTemperature = toCopy.getmaxCanopyTemperature();
        }
    }
    public Double getdiffusionLimitedEvaporation()
    { return diffusionLimitedEvaporation; }

    public void setdiffusionLimitedEvaporation(Double _diffusionLimitedEvaporation)
    { this.diffusionLimitedEvaporation= _diffusionLimitedEvaporation; } 
    
    public Double getconductance()
    { return conductance; }

    public void setconductance(Double _conductance)
    { this.conductance= _conductance; } 
    
    public Double getminCanopyTemperature()
    { return minCanopyTemperature; }

    public void setminCanopyTemperature(Double _minCanopyTemperature)
    { this.minCanopyTemperature= _minCanopyTemperature; } 
    
    public Double getmaxCanopyTemperature()
    { return maxCanopyTemperature; }

    public void setmaxCanopyTemperature(Double _maxCanopyTemperature)
    { this.maxCanopyTemperature= _maxCanopyTemperature; } 
    
}

File: src/cs/SQ_Energy_Balance/EnergyBalanceState.cs
using System;
using System.Collections.Generic;
public class EnergyBalanceState 
{
    private double _diffusionLimitedEvaporation;
    private double _conductance;
    private double _minCanopyTemperature;
    private double _maxCanopyTemperature;
    
        public EnergyBalanceState() { }
    
    
    public EnergyBalanceState(EnergyBalanceState toCopy, bool copyAll) // copy constructor 
    {
    if (copyAll)
    {
    
    diffusionLimitedEvaporation = toCopy.diffusionLimitedEvaporation;
    conductance = toCopy.conductance;
    minCanopyTemperature = toCopy.minCanopyTemperature;
    maxCanopyTemperature = toCopy.maxCanopyTemperature;
    }
    }
    public double diffusionLimitedEvaporation
        {
            get { return this._diffusionLimitedEvaporation; }
            set { this._diffusionLimitedEvaporation= value; } 
        }
    public double conductance
        {
            get { return this._conductance; }
            set { this._conductance= value; } 
        }
    public double minCanopyTemperature
        {
            get { return this._minCanopyTemperature; }
            set { this._minCanopyTemperature= value; } 
        }
    public double maxCanopyTemperature
        {
            get { return this._maxCanopyTemperature; }
            set { this._maxCanopyTemperature= value; } 
        }
}